shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1075,19 +1223,108 @@ - (CGRect)caretRectForPosition:(UITextPosition*)position {
+          initWithRectAndInfo:rect
+             writingDirection:UITextWritingDirectionNatural
+                containsStart:(i == 0)
+                  containsEnd:(i == fml::RangeForCharactersInRange(self.text,
+                                                                   NSMakeRange(0, self.text.length))
+                                        .length)
+                   isVertical:FALSE];
+      [rects addObject:selectionRect];
+    }
+  }
+  return rects;
 }
 
 - (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRange*)range {
-  // TODO(cbracken) Implement.
-  return range.start;
+  NSUInteger start = ((FlutterTextPosition*)range.start).index;
+  NSUInteger end = ((FlutterTextPosition*)range.end).index;
+
+  NSUInteger _closestIndex = 0;
+  CGRect _closestRect = CGRectZero;
+  NSUInteger _closestPosition = 0;
+  float _closestY = 0;
+  float _closestX = 0;
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    NSUInteger position = [_selectionRects[i][4] unsignedIntegerValue];
+    if (position >= start && position <= end) {
+      CGRect rect =
+          CGRectMake([_selectionRects[i][0] floatValue], [_selectionRects[i][1] floatValue],
+                     [_selectionRects[i][2] floatValue], [_selectionRects[i][3] floatValue]);
+      CGPoint pointForComparison =
+          CGPointMake(rect.origin.x, rect.origin.y + rect.size.height * 0.5);
+      float yDist = abs(pointForComparison.y - point.y);
+      float xDist = abs(pointForComparison.x - point.x);
+      if (_closestIndex == 0 ||
+          (yDist < _closestY - 1 ||
+           (yDist >= _closestY - 1 && yDist <= _closestY + 1 &&
+            ((point.y <= rect.origin.y + rect.size.height && xDist < _closestX) ||
+             (point.y > rect.origin.y + rect.size.height &&
+              rect.origin.x > _closestRect.origin.x))))) {
+        _closestY = yDist;
+        _closestX = xDist;
+        _closestIndex = i;
+        _closestRect = rect;
+        _closestPosition = position;
+      }
+    }
+  }
+
+  FlutterTextRange* textRange = [[FlutterTextRange
+      rangeWithNSRange:fml::RangeForCharactersInRange(self.text, NSMakeRange(0, self.text.length))]
+      copy];
+
+  if ([_selectionRects count] > 0 && textRange.range.length == end) {
+    NSUInteger i = [_selectionRects count] - 1;
+    NSUInteger position = [_selectionRects[i][4] unsignedIntegerValue] + 1;
+    if (position <= end) {
+      CGRect rect =
+          CGRectMake([_selectionRects[i][0] floatValue], [_selectionRects[i][1] floatValue],
+                     [_selectionRects[i][2] floatValue], [_selectionRects[i][3] floatValue]);
+      CGPoint pointForComparison =
+          CGPointMake(rect.origin.x + rect.size.width, rect.origin.y + rect.size.height * 0.5);
+      float yDist = abs(pointForComparison.y - point.y);
+      float xDist = abs(pointForComparison.x - point.x);
+      if (yDist < _closestY - 1 ||
```

Comment:
Looks like duplicated logic from above, more reason to make it a static function.